standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/Msck.java

```diff
@@ -229,102 +239,168 @@ public int repair(MsckInfo msckInfo) {
-      if (acquireLock && txnId > 0) {
-          if (success) {
-            try {
-              LOG.info("txnId: {} succeeded. Committing..", txnId);
-              getMsc().commitTxn(txnId);
-            } catch (Exception e) {
-              LOG.warn("Error while committing txnId: {} for table: {}", txnId, qualifiedTableName, e);
-              ret = 1;
-            }
-          } else {
-            try {
-              LOG.info("txnId: {} failed. Aborting..", txnId);
-              getMsc().abortTxns(Lists.newArrayList(txnId));
-            } catch (Exception e) {
-              LOG.warn("Error while aborting txnId: {} for table: {}", txnId, qualifiedTableName, e);
-              ret = 1;
-            }
-          }
+      if (txnId > 0) {
+        success = closeTxn(qualifiedTableName, success, txnId) && success;
       }
       if (getMsc() != null) {
         getMsc().close();
         msc = null;
       }
     }
+    return success ? 0 : 1;
+  }
 
+  private boolean closeTxn(String qualifiedTableName, boolean success, long txnId) {
+    boolean ret = true;
+    if (success) {
+      try {
+        LOG.info("txnId: {} succeeded. Committing..", txnId);
+        getMsc().commitTxn(txnId);
+      } catch (Exception e) {
+        LOG.warn("Error while committing txnId: {} for table: {}", txnId, qualifiedTableName, e);
+        ret = false;
+      }
+    } else {
+      try {
+        LOG.info("txnId: {} failed. Aborting..", txnId);
+        getMsc().abortTxns(Lists.newArrayList(txnId));
+      } catch (Exception e) {
+        LOG.warn("Error while aborting txnId: {} for table: {}", txnId, qualifiedTableName, e);
+        ret = false;
+      }
+    }
     return ret;
   }
 
-  private LockRequest createLockRequest(final String dbName, final String tableName) throws TException {
-    UserGroupInformation loggedInUser = null;
-    String username;
+  private void logResult(CheckResult result) {
+    LOG.info("Tables not in metastore: {}", result.getTablesNotInMs());
+    LOG.info("Tables missing on filesystem: {}", result.getTablesNotOnFs());
+    LOG.info("Partitions not in metastore: {}", result.getPartitionsNotInMs());
+    LOG.info("Partitions missing from filesystem: {}", result.getPartitionsNotOnFs());
+    LOG.info("Expired partitions: {}", result.getExpiredPartitions());
+  }
+
+  private boolean writeResultToFile(MsckInfo msckInfo, CheckResult result, List<String> repairOutput,
+      long partitionExpirySeconds) {
+    boolean success = true;
+    BufferedWriter resultOut = null;
     try {
-      loggedInUser = UserGroupInformation.getLoginUser();
+      Path resFile = new Path(msckInfo.getResFile());
+      FileSystem fs = resFile.getFileSystem(getConf());
+      resultOut = new BufferedWriter(new OutputStreamWriter(fs.create(resFile)));
+
+      boolean firstWritten = false;
+      firstWritten |= writeMsckResult(result.getTablesNotInMs(),
+        "Tables not in metastore:", resultOut, firstWritten);
```

Comment:
could we move error messages to final variables (constants)?

Reply:
fixed